File: cai/v2/cai.c
/*

	# funzioni

	# Programmi di Laboratorio di Programmazione in Linguaggio "C" e Linguaggio "Python" ("PY"). Alunno: Bucchianico Enrico Ruggiero, 4^Finf


		funzioni

		Programmi di Laboratorio di Programmazione in Linguaggio "C" e Linguaggio "Python" ("PY"). Alunno: Bucchianico Enrico Ruggiero, 4^Finf
		
		
		
			 - Directory "cai": Linguaggio "C":
			 		Fornisce aiuto ad uno studente di scuola elementare ad apprendere la moltiplicazione.
			 		- Sottodirectory "v2":Versione N° 2:
			 			- Modulo "random" ("random.h"|"random.c"): Contentente due funzioni:
			 				- "void random_init()" per l'inizializzazione del modulo.
			 				- "int random_between(int min,int max)" per la generazione di un numero pseudocasuale compreso tra min (incluso) e max (escluso).
			 			- Modulo "cai" ("cai.h"|"cai.c"): Gestisce le operazioni legate alla scelta dei numeri e alla gestione delle risposte.
			 			- File "main.c" per la verifica dei moduli ed il calcolo delle risposte corrette tra 10 date.
		 			
*/

#include <stdio.h>

#include "cai.h"

#include "random.h"

#define N_ANSWERS 4

void cai_init()
{
	random_init();
}

int cai_cifra()
{
	return random_between(0,10);
}

int cai_risposta()
{
	return random_between(0,N_ANSWERS);
}

void cai_esatta()
{
	switch(cai_risposta())
	{
		case 0:
			printf("\nVery good!\n");
			break;
		case 1:
			printf("\nExcellent!\n");
			break;
		case 2:
			printf("\nNice work!\n");
			break;
		case 3:
			printf("\nKeep up the good work!\n");
			break;
		default:
			printf("\nThere seems to be a problem: somethin gone wrong!\n");
			break;
	}
	
	putchar('\n');
	
}

void cai_errata()
{
	switch(cai_risposta())
	{
		case 0:
			printf("\nNo, Please try again.\n");
			break;
		case 1:
			printf("\nWrong. Try once more.\n");
			break;
		case 2:
			printf("\nDon't give up!\n");
			break;
		case 3:
			printf("\nNo. Keep trying.\n");
			break;
		default:
			printf("\nThere seems to be a problem: somethin gone wrong!\n");
			break;
	}
  
	putchar('\n');
  
}

File: cai/py/cai.py
'''

	# funzioni

	# Programmi di Laboratorio di Programmazione in Linguaggio "C" e Linguaggio "Python" ("PY"). Alunno: Bucchianico Enrico Ruggiero, 4^Finf


		funzioni

		Programmi di Laboratorio di Programmazione in Linguaggio "C" e Linguaggio "Python" ("PY"). Alunno: Bucchianico Enrico Ruggiero, 4^Finf
		
		
		
			 - Directory "cai":
			 		Fornisce aiuto ad uno studente di scuola elementare ad apprendere la moltiplicazione.
			 		- Sottodirectory "py": Versione N° 2: Linguaggio "Python" ("PY"):
			 			- Modulo "cai.py": Gestisce le operazioni legate alla scelta dei numeri, generati in maniera pseudocasuale - compresi tra min (incluso) e max 									    (escluso) - e le risposte.
			 			- File "main.py" per la verifica dei moduli ed il calcolo delle risposte corrette tra 10 date.
		 			
'''

import random

N_ANSWERS = 4

def cai_cifra():
	"""
		>>> from functools import reduce
		>>> cifre=[cai_cifra() for i in range(100)]
		>>> inRange=[c>=0 and c<10 for c in cifre]
		>>> allInRange=reduce(lambda x, y: x and y, inRange)
		>>> allInRange
		True
	"""
	return random.randrange(0,10)


def cai_risposta():
	"""
		>>> from functools import reduce
		>>> risposte=[cai_risposta() for i in range(100)]
		>>> inRange=[r>=0 and r<N_ANSWERS for r in risposte]
		>>> allInRange=reduce(lambda x, y: x and y, inRange)
		>>> allInRange
		True
	"""
	return random.randrange(0,N_ANSWERS)

def cai_esatta():
	risposta=cai_risposta()
	if risposta==0:
		print("\nVery good!\n")
	elif risposta==1:
		print("\nExcellent!\n")
	elif risposta==2:
		print("\nNice work!\n")
	elif risposta==3:
		print("\nKeep up the good work!\n")
	else:
		print("\nThere seems to be a problem: something gone wrong!\n")

def cai_errata():
	risposta = cai_risposta()
	if risposta == 0:
		print("\nNo, Please try again.\n")
	elif risposta == 1:
		print("\nWrong. Try once more.\n")
	elif risposta == 2:
		print("\nDon't give up!\n")
	elif risposta == 3:
		print("\nNo. Keep trying.\n")
	else:
		print("\nThere seems to be a problem: something gone wrong!\n")


if __name__=='__main__':
	import doctest
	doctest.testmod()